pynest/nest/lib/hl_api_spatial.py

```diff
@@ -842,7 +842,7 @@ def SelectNodesByMask(layer, anchor, mask_obj):
     node_id_list = sli_func('SelectNodesByMask',
                             layer, anchor, mask_datum)
 
-    return NodeCollection(node_id_list)
+    return NodeCollection(sorted(node_id_list))
```

Comment:
It might be an idea to add a comment here as well, about NodeCollections needing to be sorted.

Reply:
Done!